atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/table/common/TableTasks.java

```diff
@@ -209,8 +217,16 @@ private static void diffExternal(final DiffStrategy strategy,
                                      final DiffTask task) throws InterruptedException {
         BlockingWorkerPool pool = new BlockingWorkerPool(exec, threadCount);
         for (final MutableRange range : getRanges(threadCount, batchSize)) {
+            if (Thread.currentThread().isInterrupted()) {
```

Comment:
....wait. This looks like we have the same `for` loop twice in different parts of this class. Ideally we'd only have to write the `if` block once - not four times.